src/template.ts

```diff
@@ -1,23 +1,29 @@
 const shiki = require("shiki");
 
-const highlightCode = async (content: string) => {
+const highlightCode = async (content: string,themeName: string) => {
   const highlighter = await shiki.getHighlighter({
-    theme: "one-dark-pro", // You can specify other themes like 'dark-plus', 'light-plus', etc.
+    theme: themeName, // You can specify other themes like 'dark-plus', 'light-plus', etc.
   });
 
   const htmlContent = highlighter.codeToHtml(content, "javascript");
   return await htmlContent;
 };
 
-const generateTemplate = async (content: string) => {
+const generateTemplate = async (content: string, themeName: string) => {
+  // get background color from the theme.
+  const highlighter = await shiki.getHighlighter({
```

Comment:
This function is used twice - here and in the `highlightCode`.

return `highlighter` from `highlightCode` and do rest of the things (`codeToHtml()` and `getBackgroundColor()`) in this function.